aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,901 @@
+///
+/// An example of DAO delegation flow
+///  1. the DAO creator can delegate her DAO through `delegate_dao` to another admin address
+///
+/// An example of DAO removal.
+/// 1. The DAO creator can call `reclaim_signer_capability` to remove her DAO from the platform and get back her resource account's signercapability
+///
+/// Note: This current version stores all the signer_capability in the contract space. If the code upgrades, a potential risk is the platform operation can obtain the signer capability of the DAOs on this platform
+/// It allows the operator to obtain the signer of these DAOs without any consent.
+/// TODO: explore new tooling to dynamically compiling DAO modules on server side and store the signercapability in each DAO's own moudle to avoid exposing signer capability to platform
+module dao_platform::nft_dao {
+    use aptos_framework::account::{SignerCapability, create_signer_with_capability};
+    use aptos_framework::account;
+    use aptos_framework::aptos_account::deposit_coins;
+    use aptos_framework::coin;
+    use aptos_framework::timestamp;
+    use aptos_std::bucket_table::BucketTable;
+    use aptos_std::bucket_table;
+    use aptos_std::table::Table;
+    use aptos_std::table;
+    use aptos_token::property_map::PropertyMap;
+    use aptos_token::property_map;
+    use aptos_token::token::{Self, TokenId};
+    use dao_platform::nft_dao_events::{Self, emit_create_dao_event};
+    use std::bcs;
+    use std::error;
+    use std::option::Option;
+    use std::option;
+    use std::signer;
+    use std::string::String;
+    use std::string;
+    use std::vector;
+
+
+    /// This account doesn't have enough voting power
+    const EVOTING_POWER_NOT_ENOUGH: u64 = 1;
+
+    /// This account doesn't own this DAO's voting token
+    const ENOT_OWN_THE_VOTING_DAO_TOKEN: u64 = 2;
+
+    /// This function is not supported in proposal
+    const ENOT_SUPPROTED_FUNCTION: u64 = 3;
+
+    /// Can only propose a start time in future
+    const EPROPOSAL_START_TIME_SHOULD_BE_IN_FUTURE: u64 = 4;
+
+    /// Invalid admin account
+    const EINVALID_ADMIN_ACCOUNT: u64 = 5;
+
+    /// String length exceeds limits
+    const ESTRING_TOO_LONG: u64 = 6;
+
+    /// Proposal ended and no more voting
+    const EPROPOSAL_ENDED: u64 = 7;
+
+    /// Proposal has not started voting
+    const EPROPOSAL_NOT_STARTED: u64 = 8;
+
+    /// Proposal has already been resolved
+    const EPROPOSAL_RESOLVED: u64 = 9;
+
+    /// Token already voted for the proposal
+    const ETOKEN_ALREADY_VOTED: u64 = 10;
+
+    /// DAO doesn't exist at this address
+    const EDAO_NOT_EXIST: u64 = 11;
+
+    /// Proposal resource not created
+    const EPRPOSALS_NOT_EXIST_AT_ADDRESS: u64 = 12;
+
+    /// Proposal with specified id doesn't exist
+    const EPRPOSAL_ID_NOT_EXIST: u64 = 13;
+
+    /// DAO already created at this address
+    const EDAO_ADDRESS_USED: u64 = 14;
+
+    /// Already an admin of the DAO
+    const EALREADY_ADMIN_OF_THE_DAO: u64 = 15;
```

Comment:
This error doesn't seem to be used. Can you delete any that are not used?

Reply:
removed